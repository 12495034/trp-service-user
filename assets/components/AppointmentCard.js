import React from 'react'
import { View, StyleSheet, Pressable } from 'react-native'
import { Text } from 'react-native-paper';

import CalledIcon from '../Icons/CalledIcon'
import CheckInIcon from '../Icons/CheckInIcon'
import TestCompleteIcon from '../Icons/TestCompleteIcon'
import { AlertCancel } from '../functions/generalFunctions/AlertCancel'
import { handleAlertInformation } from '../functions/generalFunctions/Alerts'
import { createDateString } from '../functions/generalFunctions/createDateString'

/**
 * Pressable Appointment Card component rendered within on the Appointments Screen
 */
export default function AppointmentCard(props) {
    return (
        <Pressable
            style={({ pressed }) => [
                { backgroundColor: pressed ? '#F7C3E9' : '#0000', borderRadius: 5 },
            ]}
            onLongPress={() => {
                if (props.isCancellable && props.status === "Active") {
                    AlertCancel(
                        props.status,
                        props.checkedIn,
                        "Cancel Appointment",
                        "Do you wish to cancel this appointment?",
                        "Yes", "No",
                        () => props.cancel(props.slot, props.time, props.clinicId, props.userId))
                } else if (props.status === "Active") {
                    handleAlertInformation("Cancel Appointment", "You are unable to cancel this appointment as it is less than 24 hrs until you are due to attend. If you are not able to make the appointment please inform the rainbow project on 02890 319030 or you will be marked as un-attended")
                } else {
                    //do nothing
                }

            }}
        >
            <View style={styles.card}>
                <View style={styles.col1}>
                    <Text>{props.location}</Text>
                    <Text>{props.center}</Text>
                    <Text>{props.addDetails}</Text>
                    <Text>{props.clinicAddress}</Text>
                    <Text>{props.clinicPostcode}</Text>
                </View>
                <View style={styles.col2}>
                    <Text>{createDateString(props.date)}</Text>
                    <Text >Slot {props.slot} @ {props.time}</Text>
                    <Text style={styles.hilight}>{props.tester}</Text>
                    <Text></Text>
                    <Text>{props.status}</Text>

                </View>
                <View style={styles.col3} onTouchStart={props.userCheckIn}>
                    <CheckInIcon checkedIn={props.checkedIn} />
                    <CalledIcon checkedIn={props.checkedIn} called={props.called} />
                    <TestCompleteIcon checkedIn={props.checkedIn} complete={props.wasSeen} />
                </View>
            </View>
            <View>

            </View>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    card: {
        width: '100%',
        paddingTop: 10,
        paddingBottom: 10,
        flexDirection: 'row',
        justifyContent: 'space-around',
        borderColor: 'black',
        borderStyle: 'solid',
        borderWidth: 1,
        borderRadius: 5,

    },
    col1: {
        flexDirection: 'column',

    },
    col2: {
        flexDirection: 'column',
    },
    col3: {
        flexDirection: 'column',
        justifyContent: 'space-between',
    },
    hilight: {
        color: 'red'
    }
})
