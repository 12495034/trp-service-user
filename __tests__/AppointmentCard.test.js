import React from 'react';
import renderer from 'react-test-renderer';
import AppointmentCard from '../assets/components/AppointmentCard';

describe('Appointment card tests', () => {
    test('renders AppointmentCard component correctly', () => {
        const tree = renderer.create(<AppointmentCard
            clinicId={123456}
            tester={'Leo Lardie'}
            location={'Belfast'}
            center={'LGBT Center'}
            date={'3000-01-01'}
            time={'18:00'}
            slot={1}
            checkedIn={true}
            called={true}
            wasSeen={true}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});