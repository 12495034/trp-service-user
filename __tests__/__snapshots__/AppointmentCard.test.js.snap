// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment card tests renders AppointmentCard component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#0000",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      {
        "borderColor": "black",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        Belfast
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        LGBT Center
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        Wed Jan 01 3000
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        Slot 
        1
         @ 
        18:00
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "red",
            },
          ]
        }
      >
        Leo Lardie
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "green",
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "purple",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰂚
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "green",
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View />
</View>
`;
